Fails to open project on EAP/Beta

When starting Idea and selecting a project from the welcome screen, I get the following error and the project doesn't open.

```
java.lang.RuntimeException: Cannot invoke (class=ProjectListener, method=projectOpened, topic=ProjectManagerListener)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeListener(MessageBusImpl.kt:639)
	at com.intellij.util.messages.impl.MessageBusImplKt.executeOrAddToQueue(MessageBusImpl.kt:466)
	at com.intellij.util.messages.impl.ToDirectChildrenMessagePublisher.publish$intellij_platform_core(CompositeMessageBus.kt:259)
	at com.intellij.util.messages.impl.MessagePublisher.invoke(MessageBusImpl.kt:421)
	at jdk.proxy2/jdk.proxy2.$Proxy90.projectOpened(Unknown Source)
	at com.intellij.openapi.project.impl.ProjectManagerImplKt$runInitProjectActivities$$inlined$launchAndMeasure$1.invokeSuspend(startUpMeasurer.kt:48)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at com.intellij.openapi.application.impl.DispatchedRunnable.run(DispatchedRunnable.kt:35)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:209)
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:21)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:191)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:813)
	at com.intellij.openapi.application.impl.ApplicationImpl$3.run(ApplicationImpl.java:429)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:74)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:114)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:36)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:779)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:730)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:724)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:749)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:909)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:756)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$5(IdeEventQueue.java:437)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:774)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:436)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:105)
	at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:615)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:434)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:813)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:480)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
Caused by: java.util.NoSuchElementException: Array is empty.
	at kotlin.collections.ArraysKt___ArraysKt.first(_Arrays.kt:1012)
	at io.unthrottled.amii.tools.BalloonTools.getIDEFrame(BalloonTools.kt:24)
	at io.unthrottled.amii.listeners.IdleEventListener.<init>(IdleEventListener.kt:29)
	at io.unthrottled.amii.ProjectListeners.<init>(PluginMaster.kt:93)
	at io.unthrottled.amii.PluginMaster.registerListenersForProject(PluginMaster.kt:51)
	at io.unthrottled.amii.PluginMaster.projectOpened(PluginMaster.kt:42)
	at io.unthrottled.amii.listeners.ProjectListener.projectOpened(ProjectListener.kt:12)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeMethod(MessageBusImpl.kt:655)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeListener(MessageBusImpl.kt:625)
	... 39 more
```

IntelliJ IDEA 2022.3 Beta (Ultimate Edition)
Build #IU-223.7401.7, built on November 2, 2022
Runtime version: 17.0.5+1-b653.6 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 10 10.0
Kotlin: 223-1.7.20-release-201-IJ7401.7

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,12 @@
 
 # AMII Changelog
 
+## [1.1.4]
+
+### Fixed
+
+- Not being able to open any project from the Welcome Screen on the 2022.3 Beta build.
+
 ## [1.1.3]
 
 ### Fixed

--- a/docs/RELEASE-NOTES.md
+++ b/docs/RELEASE-NOTES.md
@@ -1,14 +1,3 @@
-### Added
-
-- Custom Content, please <a href="https://github.com/ani-memes/AMII/tree/main#custom-assets">
-  documentation</a> for more details!
-- 2022.3 Build Support
-
 ### Fixed
 
-- `Show Previous Meme` not replaying sound for assets that have sound.
-- Reactions when debugging Flutter tests on Windows.
-- Issue
-  preventing [Apex unit tests](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_dev_guide.htm)
-  reactions from showing up
-  when [using the Illuminated Cloud 2](https://plugins.jetbrains.com/plugin/10253-illuminated-cloud-2) plugin.
+- Not being able to open any project from the Welcome Screen on the 2022.3 Beta build.

--- a/gradle.properties
+++ b/gradle.properties
@@ -3,7 +3,7 @@
 
 pluginGroup = io.unthrottled
 pluginName_ = Anime Memes
-pluginVersion = 1.1.3
+pluginVersion = 1.1.4
 pluginSinceBuild = 203.7148.57
 pluginUntilBuild = 223.*
 # Plugin Verifier integration -> https://github.com/JetBrains/gradle-intellij-plugin#plugin-verifier-dsl
@@ -15,10 +15,10 @@ platformVersion = 2022.2.1
 platformDownloadSources = true
 # Plugin Dependencies -> https://www.jetbrains.org/intellij/sdk/docs/basics/plugin_structure/plugin_dependencies.html
 # Example: platformPlugins = com.intellij.java,com.jetbrains.php:203.4449.22
-platformPlugins = io.acari.DDLCTheme:88.1-1.4.0,NodeJS,Dart:222.3739.24,io.flutter:70.0.5
+platformPlugins = io.acari.DDLCTheme:88.1-1.5.4,NodeJS,Dart:222.3739.24,io.flutter:70.0.5
 
 idePath=
-#idePath=/Users/alexsimons/Library/Application Support/JetBrains/Toolbox/apps/WebStorm/ch-0
+#idePath=/Users/alexsimons/Library/Application Support/JetBrains/Toolbox/apps/WebStorm/ch-0/223.7401.14/WebStorm 2022.3 EAP.app/Contents
 #idePath=/home/alex/.local/share/JetBrains/Toolbox/apps/AndroidStudio/ch-0/203.7321754
 #idePath=/home/alex/.local/share/JetBrains/Toolbox/apps/PyCharm-P/ch-0/211.6085.15
 

--- a/src/main/kotlin/icons/AMIIIcons.kt
+++ b/src/main/kotlin/icons/AMIIIcons.kt
@@ -34,6 +34,7 @@ object AMIIIcons {
   val E1F612 = IconLoader.getIcon("/icons/emojis/1f612.svg", javaClass)
   val E1F620 = IconLoader.getIcon("/icons/emojis/1f620.svg", javaClass)
   val E1F632 = IconLoader.getIcon("/icons/emojis/1f632.svg", javaClass)
+  @JvmField
   val E1F634 = IconLoader.getIcon("/icons/emojis/1f634.svg", javaClass)
   val E1F642 = IconLoader.getIcon("/icons/emojis/1f642.svg", javaClass)
   val E1F973 = IconLoader.getIcon("/icons/emojis/1f973.svg", javaClass)

--- a/src/main/kotlin/io/unthrottled/amii/actions/FireIdleAction.kt
+++ b/src/main/kotlin/io/unthrottled/amii/actions/FireIdleAction.kt
@@ -1,0 +1,26 @@
+package io.unthrottled.amii.actions
+
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.project.DumbAware
+import io.unthrottled.amii.events.EVENT_TOPIC
+import io.unthrottled.amii.events.UserEvent
+import io.unthrottled.amii.events.UserEventCategory
+import io.unthrottled.amii.events.UserEvents
+import io.unthrottled.amii.tools.PluginMessageBundle
+
+class FireIdleAction : AnAction(), DumbAware {
+  override fun actionPerformed(e: AnActionEvent) {
+    val project = e.project
+    project?.messageBus
+      ?.syncPublisher(EVENT_TOPIC)
+      ?.onDispatch(
+        UserEvent(
+          UserEvents.IDLE,
+          UserEventCategory.NEUTRAL,
+          PluginMessageBundle.message("user.event.idle.name"),
+          project
+        )
+      )
+  }
+}

--- a/src/main/kotlin/io/unthrottled/amii/listeners/IdleEventListener.kt
+++ b/src/main/kotlin/io/unthrottled/amii/listeners/IdleEventListener.kt
@@ -26,7 +26,9 @@ import java.util.concurrent.TimeUnit
 class IdleEventListener(private val project: Project) : Runnable, Disposable, AWTEventListener {
   private val messageBus = ApplicationManager.getApplication().messageBus.connect()
   private val log = Logger.getInstance(this::class.java)
-  private val rootPane = BalloonTools.getIDEFrame(project).component
+  private val rootPane = lazy {
+    BalloonTools.getIDEFrame(project).component
+  }
   private var idleTimeout =
     TimeUnit.MILLISECONDS.convert(
       getCurrentTimoutInMinutes(),
@@ -81,14 +83,14 @@ class IdleEventListener(private val project: Project) : Runnable, Disposable, AW
     idleAlarm.addRequest(this, idleTimeout)
   }
 
-  private val allowedMouseEvents = setOf(
-    MouseEvent.MOUSE_PRESSED,
-    MouseEvent.MOUSE_CLICKED,
-  )
-  override fun eventDispatched(e: AWTEvent) {
-    if (e !is InputEvent || !UIUtil.isDescendingFrom(e.component, rootPane)) return
+  private val allowedMouseEvents =
+    MouseEvent.MOUSE_PRESSED or
+      MouseEvent.MOUSE_CLICKED
 
-    if (e is MouseEvent && allowedMouseEvents.contains(e.id).not()) return
+  override fun eventDispatched(e: AWTEvent) {
+    if (e !is InputEvent || !UIUtil.isDescendingFrom(e.component, rootPane.value)) return
+
+    if (e is MouseEvent && (allowedMouseEvents and e.id) != e.id) return
 
     idleAlarm.cancelAllRequests()
     idleAlarm.addRequest(this, idleTimeout)

--- a/src/main/kotlin/io/unthrottled/amii/memes/MemePanel.kt
+++ b/src/main/kotlin/io/unthrottled/amii/memes/MemePanel.kt
@@ -105,12 +105,11 @@ class MemePanel(
     private const val WHITE_HEX = 0x00FFFFFF
     private const val TENTH_OF_A_SECOND_MULTIPLICAND = 100
 
-    private val ALLOWED_KEYS = setOf(
-      KeyEvent.VK_SHIFT,
-      KeyEvent.VK_CONTROL,
-      KeyEvent.VK_ALT,
-      KeyEvent.VK_META,
-    )
+    private const val NON_FOCUS_LOSS_KEYS =
+      KeyEvent.VK_SHIFT or
+        KeyEvent.VK_CONTROL or
+        KeyEvent.VK_ALT or
+        KeyEvent.VK_META
   }
 
   private var alpha = 0.0f
@@ -199,7 +198,7 @@ class MemePanel(
       ) {
         if (
           isFocusLoss &&
-          ALLOWED_KEYS.contains(event.keyCode).not()
+          (NON_FOCUS_LOSS_KEYS and event.keyCode) != event.keyCode
         ) {
           dismissMeme()
         }

--- a/src/main/kotlin/io/unthrottled/amii/onboarding/UpdateNotification.kt
+++ b/src/main/kotlin/io/unthrottled/amii/onboarding/UpdateNotification.kt
@@ -22,11 +22,7 @@ private fun buildUpdateMessage(updateAsset: String): String =
   """
       What's New?<br>
       <ul>
-        <li>Added Custom Content Support.</li>
-        <li>Replay Previous Meme Replays Sound Now.</li>
-        <li>Fixed Apex Unit Test Reactions</li>
-        <li>Fixed Flutter Unit Test Reactions</li>
-        <li>2022.3 Build Support</li>
+        <li>Fixed startup in 2022.3 build</li>
       </ul>
       <br>See the <a href="https://github.com/ani-memes/AMII#documentation">documentation</a> for features, usages, and configurations.
       <br>The <a href="https://github.com/ani-memes/AMII/blob/master/CHANGELOG.md">changelog</a> is available for more details.

--- a/src/main/resources/META-INF/plugin.xml
+++ b/src/main/resources/META-INF/plugin.xml
@@ -102,6 +102,12 @@
               description="Action calms down MIKU">
         <keyboard-shortcut keymap="$default" first-keystroke="ctrl alt R"/>
       </action>
+      <action id="amii.idle"
+              class="io.unthrottled.amii.actions.FireIdleAction"
+              text="Dispatch Idle Event"
+              icon="AMIIIcons.E1F634"
+              description="Tells MIKU to start an Idle event">
+      </action>
       <action id="amii.clear"
               class="io.unthrottled.amii.actions.ClearAction"
               text="Clear Memes"

